tests_e2e/playwright/navigator.py

```diff
@@ -0,0 +1,448 @@
+        self.slow_mo = slow_mo
+        self.browser = browser
+        self.instance_name = instance_name
+        self.video_dir = video_dir
+
+        self.setup(
+            browser=self.browser,
+            headless=self.headless,
+            slow_mo=self.slow_mo,
+        )
+        self.wait_for_server_spinup = 30000  # 5 * 60 * 1000  # 5 minutes in ms
+
+    @property
+    def initialize(self):
+        """Ensure that the Navigator is setup and ready for testing."""
+        if not self.initialized:
+            self.setup(
+                browser=self.browser,
+                headless=self.headless,
+                slow_mo=self.slow_mo,
+            )
+
+    def setup(self, browser, headless, slow_mo):
+        """Initial setup for running playwright. Starts playwright, creates
+        the browser object, a new browser context, and a new page object.
+
+        Parameters
+        ----------
+        browser: str
+            Browser on which to run tests. Options are "chromium",
+            "webkit", and "firefox".
+        headless: bool
+            Run the tests in headless mode (without visuals) if True
+        slow_mo: int
+            Additional milliseconds to add to each Playwright command,
+            creating the effect of running the tests in slow motion so they are
+            easier for humans to follow.
+        """
+        logger.debug(">>> Setting up browser for Playwright")
+
+        self.playwright = sync_playwright().start()
+        if browser == "chromium":
+            self.browser = self.playwright.chromium.launch(
+                headless=headless, slow_mo=slow_mo
+            )
+        elif browser == "webkit":
+            self.browser = self.playwright.webkit.launch(
+                headless=headless, slow_mo=slow_mo
+            )
+        elif browser == "firefox":
+            self.browser = self.playwright.firefox.launch(
+                headless=headless, slow_mo=slow_mo
+            )
+        else:
+            raise RuntimeError(f"{browser} browser is not recognized.")
+        self.context = self.browser.new_context(
+            ignore_https_errors=True,
+            record_video_dir=self.video_dir,
+        )
+        self.page = self.context.new_page()
+        self.initialized = True
+
+    def teardown(self) -> None:
+        """Shut down and close playwright. This is important to ensure that
+        no leftover processes are left running in the background."""
+        self.context.close()
+        self.browser.close()  # Make sure to close, so that videos are saved.
+        self.playwright.stop()
+        logger.debug(">>> Teardown complete.")
+
+    def login(self) -> None:
+        """Login to nebari deployment using the auth method on the class."""
+        if self.auth == "google":
+            self.login_google()
+        elif self.auth == "password":
+            self.login_password()
+        else:
+            raise ValueError(f"Auth type of {self.auth} is invalid.")
```

Comment:
Nit: since we expect to add more login methods here, let's make this more extensible from the start:

```suggestion
        try:
            return {
                "google": self.login_google,
                "password": self.login_password,
            }()
        except KeyError:
            raise ValueError(f"Auth type of {self.auth} is invalid.") from None
```